src/octoprint/util/comm.py

```diff
@@ -840,6 +843,9 @@ def startPrint(self, pos=None):
 
 				self.resetLineNumbers()
 
+				from flask.ext.login import current_user
```

Comment:
See my other comment - this won't work like that in all cases. Instead you should add a new parameter to the the start print methods that allows to forward a user id (which should however stay optional for backwards compatibility) which should then be attached to the currently selected file information in printer and comm layer.

``current_user`` won't always be set since you won't always have a request context here. In any case, nothing past the REST API implementation should ever depend on flask stuff if at all possible. Backend modules shouldn't need to know about frontend facing API implementations.

Reply:
Thanks for your reply. I thinked about api connections, but I not found how always get the good user wich lunch printing (by api, flask or anything else). The goal of this data in events objects is to capture the "initiator" user of the print action. It usefull for history plugin or a future printing service management (eg fablab). The canceler user could be also a good information. In my pr user field is potentially different of print_user

Reply:
I think I didn't explain myself well.

I wholeheartedly agree that adding the user who started (or cancelled or otherwise interacted with) a job to the corresponding events is a good idea! The problem here isn't adding the user, it's HOW you add it, by using the `current_user` implemented from flask. This field is only available in the context of an active request, NOT when an event gets triggered asynchronously through the system.

My point is, you'll have to add a userid parameter to the methods in question and populate that accordingly throughout the system, not just rely on `current_user` being available because it won't be in a lot of cases where print related events will be generated. 

Take this example.

User "foo" starts a print job. `current_user` is available when he does that and so `PrintStarted` gets fired with "foo" attached as user. The print continues for a while. Then the firmware sends a [pause action command](), making the print pause. A `PrintPaused` event is triggered, but `current_user` is not set (no request active). So now you have a problem.

Additionally you are tightly coupling `PrinterInterface`, `StandardPrinter` and even the `MachineComm` class to the technology of the *frontend facing API*. That's a strong code smell. It increases maintenance overhead, makes the code more tricky to test and so on. The printer object and even more so the communication layer shouldn't have to know that it gets called from a flask based API and that their might be a user stored in flask's `current_user`. It should simply get told "hey, user with id 'foo' started/paused/resumed/cancelled" this print, not try to determine that itself based on implementation details of other system components it should have no deeper knowledge of.